app/javascript/packages/document-capture/components/status-message.tsx

```diff
@@ -1,25 +1,19 @@
-/** @typedef {import('react').ReactNode} ReactNode */
+import type { ReactNode } from 'react';
 
-/**
- * @enum {string}
- */
-export const Status = {
-  ERROR: 'ERROR',
-  SUCCESS: 'SUCCESS',
-};
-
-/**
- * @typedef StatusMessageProps
- *
- * @prop {Status} status
- * @prop {string=} className
- * @prop {ReactNode=} children
- */
+export enum Status {
+  ERROR = 'ERROR',
+  SUCCESS = 'SUCCESS',
+}
 
-/**
- * @param {StatusMessageProps} props
- */
-function StatusMessage({ status, className, children }) {
+function StatusMessage({
+  status,
+  className,
+  children,
+}: {
+  status: Status;
+  className?: string;
+  children?: ReactNode;
+}) {
```

Comment:
I personally like having a named `interface StatusMessageProps` like you had prior to f3cee1f5d2460458061fb3f9e29494a5572df4be, since otherwise I think the arguments section becomes symbol soup, but it's a small detail.